File: Solutions/Go/Accepted/1833-maximum-ice-cream-bars.go
/*
Problem Title: 1833 - Maximum Ice Cream Bars
Lang: Go
Runtime: 972 ms
Memory: 8.6 MB
Problem Url: https://leetcode.com/problems/maximum-ice-cream-bars
Submission Url: https://leetcode.com/problems/maximum-ice-cream-bars/submissions/872289581
*/

func maxIceCream(costs []int, coins int) int {
    sort.Sort(sort.IntSlice(costs))

    var answer int = 0
    for _, cost := range costs {
        fmt.Println(cost)
        if cost <= coins {
            answer += 1
            coins -= cost
        }
    }

    return answer;
}

File: Solutions/C++/Accepted/1833-maximum-ice-cream-bars.cpp
/*
Problem Title: 1833 - Maximum Ice Cream Bars
Lang: C++
Runtime: 477 ms
Memory: 76.6 MB
Problem Url: https://leetcode.com/problems/maximum-ice-cream-bars
Submission Url: https://leetcode.com/problems/maximum-ice-cream-bars/submissions/872280189
*/

class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(costs.begin(), costs.end());
        int ans = 0;
        for(int cost : costs){
            if(cost <= coins){
                ans += 1;
                coins -= cost;
            }
        }
        return ans;
    }
};